File: java-sample-app/src/main/java/com/aminography/interceptor/sample/MainActivity.java
package com.aminography.interceptor.sample;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.aminography.interceptor.EditTextInterceptor;

import org.jetbrains.annotations.NotNull;

/**
 * @author aminography
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText editText = findViewById(R.id.editText);
        final TextView textView = findViewById(R.id.textView);
        final TextView typingTextView = findViewById(R.id.typingTextView);

        editText.addTextChangedListener(new EditTextInterceptor(500) {
            @Override
            public void onInterceptText(@NotNull String text, boolean isTyping) {
                typingTextView.setVisibility(isTyping ? View.VISIBLE : View.INVISIBLE);
                if (!isTyping) {
                    textView.setText(String.format("%s\n➠ %s", textView.getText(), text).trim());
                }
            }
        });
    }

}

File: sample-app/src/main/java/com/aminography/interceptor/sample/MainActivity.kt
package com.aminography.interceptor.sample

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.aminography.interceptor.EditTextInterceptor
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author aminography
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editText.addTextChangedListener(object : EditTextInterceptor(500) {
            override fun onInterceptText(text: String, isTyping: Boolean) {
                typingTextView.visibility = if (isTyping) View.VISIBLE else View.INVISIBLE
                if (!isTyping) {
                    textView.text = String.format("%s\n➠ %s", textView.text, text).trim()
                }
            }
        })
    }

}
